foo_uie_console/main.cpp

```diff
@@ -182,6 +188,14 @@ void ConsoleWindow::set_edge_style(EdgeStyle edge_style)
     }
 }
 
+void ConsoleWindow::set_append_newline(bool append_newline)
+{
+    m_append_newline = append_newline;
+    cfg_last_append_newline = static_cast<int32_t>(append_newline);
```

Comment:
The static cast looks like it can go:

```suggestion
    cfg_last_append_newline = append_newline;
```